test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1232,15 +1225,15 @@ class test_ucp_am_nbx_closed_ep : public test_ucp_am_nbx_reply {
 
         if (rx_expected) {
             request_wait(sreq);
-            wait_for_flag(&m_am_received);
+            wait_for_value(&m_recv_counter, m_send_counter);
+            EXPECT_EQ(m_recv_counter, m_send_counter);
         } else {
             // Send request may complete with error
             // (rndv should complete with EP_TIMEOUT)
             scoped_log_handler wrap_err(wrap_errors_logger);
             request_wait(sreq);
+            EXPECT_LT(m_recv_counter, m_send_counter);
```

Comment:
EXPECT_EQ(0, m_recv_counter)

Reply:
I can change it because it's non common code. Will be more clear, although the conditions has the same logic